sql/catalyst/src/main/scala/org/apache/spark/sql/errors/SqlScriptingException.scala

```diff
@@ -17,40 +17,77 @@
 
 package org.apache.spark.sql.errors
 
-import org.apache.spark.SparkException
+import org.apache.spark.{SparkException, SparkThrowableHelper}
+import org.apache.spark.sql.catalyst.trees.Origin
+import org.apache.spark.sql.errors.SqlScriptingException.errorMessageWithLineNumber
+
+class SqlScriptingException protected (
+    origin: Origin,
+    errorClass: String,
+    cause: Throwable,
+    messageParameters: Map[String, String] = Map.empty)
+  extends SparkException(
+    message = errorMessageWithLineNumber(origin, errorClass, messageParameters),
+    errorClass = Option(errorClass),
+    cause = cause,
+    messageParameters = messageParameters
+  ) {
+
+}
 
 /**
  * Object for grouping error messages thrown during parsing/interpreting phase
  * of the SQL Scripting Language interpreter.
  */
-private[sql] object SqlScriptingErrors extends QueryErrorsBase {
+private[sql] object SqlScriptingException {
```

Comment:
I'm thinking maybe we should leave `SqlScriptingErrors` instead of using `SqlScriptingException` object - if you look at the `SparkException` as an example, they use object only for some internal stuff and exceptions, parameter formatting, etc. while various error classes are used to construct concrete `SparkExceptions`.

In such case, I would move `SqlScriptingException` to something like `org.apache.spark.sql.exceptions` - I know it doesn't exist at the moment, but I think we should create the package since I don't see any more meaningful place to put the new exception type.